code/modules/admin/verbs/debug.dm

```diff
@@ -654,18 +654,10 @@ But you can call procs that are of type /mob/living/carbon/human/proc/ for that
 		exists[L.ruin_template] = landmark
 
 	var/list/names = list()
-	names += "---- Space Ruins ----"
-	for(var/name in SSmapping.space_ruins_templates)
-		names[name] = list(SSmapping.space_ruins_templates[name], ZTRAIT_SPACE_RUINS, list(/area/space))
-	names += "---- Lava Ruins ----"
-	for(var/name in SSmapping.lava_ruins_templates)
-		names[name] = list(SSmapping.lava_ruins_templates[name], ZTRAIT_LAVA_RUINS, list(/area/lavaland/surface/outdoors/unexplored))
-	names += "---- Ice Ruins ----"
-	for(var/name in SSmapping.ice_ruins_templates)
-		names[name] = list(SSmapping.ice_ruins_templates[name], ZTRAIT_ICE_RUINS, list(/area/icemoon/surface/outdoors/unexplored, /area/icemoon/underground/unexplored))
-	names += "---- Ice Underground Ruins ----"
-	for(var/name in SSmapping.ice_ruins_underground_templates)
-		names[name] = list(SSmapping.ice_ruins_underground_templates[name], ZTRAIT_ICE_RUINS_UNDERGROUND, list(/area/icemoon/underground/unexplored))
+	for(var/theme in SSmapping.themed_ruins)
+		names += "---- [theme] ----"
+		for(var/name in SSmapping.themed_ruins[theme])
+			names[name] = list(SSmapping.themed_ruins[theme][name], ZTRAIT_SPACE_RUINS, list(/area/space))
```

Comment:
What about these secondary seedRuins paramaters here, they are gone now.

Reply:
i don't know what you mean, all of these are used in `seedRuins`

Reply:
In the old spawn ruin, required zlevel trait and area type differ by ruin type. Now it will try to spawn all of them in space.

Reply:
oh im fucking dumb